framework/assets/yii.js

```diff
@@ -284,67 +274,185 @@ window.yii = (function ($) {
             }
 
             var pairs = url.substring(pos + 1).split('#')[0].split('&'),
-                params = {},
-                pair,
-                i;
+                params = {};
 
-            for (i = 0; i < pairs.length; i++) {
-                pair = pairs[i].split('=');
+            for (var i = 0, len = pairs.length; i < len; i++) {
+                var pair = pairs[i].split('=');
                 var name = decodeURIComponent(pair[0].replace(/\+/g, '%20'));
                 var value = decodeURIComponent(pair[1].replace(/\+/g, '%20'));
-                if (name.length) {
-                    if (params[name] !== undefined) {
-                        if (!$.isArray(params[name])) {
-                            params[name] = [params[name]];
-                        }
-                        params[name].push(value || '');
-                    } else {
-                        params[name] = value || '';
+                if (!name.length) {
+                    continue;
+                }
+                if (params[name] === undefined) {
+                    params[name] = value || '';
+                } else {
+                    if (!$.isArray(params[name])) {
+                        params[name] = [params[name]];
                     }
+                    params[name].push(value || '');
                 }
             }
             return params;
         },
 
         initModule: function (module) {
-            if (module.isActive === undefined || module.isActive) {
-                if ($.isFunction(module.init)) {
-                    module.init();
-                }
-                $.each(module, function () {
-                    if ($.isPlainObject(this)) {
-                        pub.initModule(this);
-                    }
-                });
+            if (module.isActive !== undefined && !module.isActive) {
+                return;
             }
+            if ($.isFunction(module.init)) {
+                module.init();
+            }
+            $.each(module, function () {
+                if ($.isPlainObject(this)) {
+                    pub.initModule(this);
+                }
+            });
         },
 
         init: function () {
             initCsrfHandler();
             initRedirectHandler();
-            initScriptFilter();
+            initAssetFilters();
             initDataMethods();
+        },
+
+        // Utility functions
```

Comment:
- Added 2 utility functions, both of them need to be publicly accesible in order to create the stubs.
- `getHostInfo` function name was not the best name for what it returns. Besides the host info it also returns protocol. `getBaseCurrentUrl` is a much intuitive IMO.